spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/PredicatePushDown.scala

```diff
@@ -0,0 +1,332 @@
+package org.apache.spark.sql.cassandra
+
+import com.datastax.spark.connector.NamedColumnRef
+import com.datastax.spark.connector.cql.TableDef
+import com.datastax.spark.connector.rdd.CassandraRDD
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.sources
+import org.apache.spark.sql.types.DataType
+
+class PredicatePushDown (predicates: Seq[Expression], tableDef: TableDef)
```

Comment:
Also, it would be nice if you added API docs for the top classes: `PushDown` and `RddSelectPdTrf`, describing their purpose